tools/schemacode/bidsschematools/render.py

```diff
@@ -469,11 +470,55 @@ def make_filename_template(
         )
 
     codeblock = codeblock.expandtabs(4)
+    codeblock = append_filename_template_legend(codeblock, pdf_format)
     codeblock = codeblock.replace("SPEC_ROOT", get_relpath(src_path))
 
     return codeblock
 
 
+def append_filename_template_legend(
+    text=None,
+    pdf_format=False,
+):
+
+    if text is None:
+        return
+
```

Comment:
I don't see a point in allowing `text` to be `None` and having a special carveout. We take a `str`, return a `str`.

```suggestion
def append_filename_template_legend(text, pdf_format=False):
```

Reply:
ha yeah this me wrongly assuming I was the cause of this [bug](https://github.com/bids-standard/bids-specification/pull/1219#issuecomment-1217769662) and trying things around but forgetting to delete them.

Reply:
FYI I a have no idea what this means:

> having a special carveout

Reply:
A carveout is a case that needs to be handled specifically; here we had an invalid value being checked for but also made the default.

Apologies, this is probably an Americanism, as it is often used to describe exemptions that are written into laws.

Reply:
No worries
I thought it was some pythonic or well known coding lingo and google was not helpful there.